dashboard/app/jobs_test.go

```diff
@@ -350,3 +351,45 @@ func TestJobRestrictedManager(t *testing.T) {
 	c.expectEQ(pollResp.Manager, build.Manager)
 	c.expectEQ(pollResp.KernelRepo, "git://restricted.git/restricted.git")
 }
+
+// Test that JobBisectFix is returned only after 30 days
+func TestBisectFixJob(t *testing.T) {
+	c := NewCtx(t)
+	defer c.Close()
+
+	// Upload a crash report
+	build := testBuild(1)
+	c.client2.UploadBuild(build)
+	crash := testCrashWithRepro(build, 1)
+	c.client2.ReportCrash(crash)
+	c.client2.pollEmailBug()
+
+	// Receive the JobBisectCause
+	resp := c.client2.pollJobs(build.Manager)
+	c.client2.expectNE(resp.ID, "")
+	c.client2.expectEQ(resp.Type, dashapi.JobBisectCause)
+	done := &dashapi.JobDoneReq{
+		ID:    resp.ID,
+		Error: []byte("testBisectFixJob:JobBisectCause"),
+	}
+	c.client2.expectOK(c.client2.JobDone(done))
+
+	// Ensure no more jobs
+	resp = c.client2.pollJobs(build.Manager)
+	c.client2.expectEQ(resp.ID, "")
+
+	// Advance time by 30 days and read out any notification emails
+	c.advanceTime(30 * 24 * time.Hour)
+	c.client2.pollEmailBug()
```

Comment:
What are these 2 emails about? Please add comments for email contents, or better some checks on Subject/Body (no need to check full body b/c it's usually hard and should be covered by other tests, but at least some contents substring).
External communication is the most important part of the system, it's good to keep it under control.

Reply:
These emails are notification emails about the bug. I've modified the test to have some more context.